### PR TITLE
Examining your uplink gives you the code

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
If you're the "owner" of an uplink, you can examine it to get the code and possible failsafe code.
## Why It's Good For The Game
It can feel slow to pull up the traitor info menu when you forget what your code is
## Changelog
:cl:
qol: examine your uplink for code
/:cl:

### CODE REVIEW

File: code/datums/components/uplink.dm [hunk 1]
```diff
@@ -128,6 +129,15 @@
...
 		load_tc(user, item)
 
+/datum/component/uplink/proc/on_examine(datum/souuce, mob/user, list/examine_list)
+	if(user != owner)
+		return
+	examine_list += span_warning("\The [parent] contains your hidden uplink.")
+	if(unlock_code)
+		examine_list += span_warning("The code to unlock it is [unlock_code].")
+	if(failsafe_code)
+		examine_list += span_warning("The failsafe code is [failsafe_code]")
```
Comment (dragomagol) [inline code/datums/components/uplink.dm:139]:
```suggestion
		examine_list += span_warning("The failsafe code is [failsafe_code].")
```

File: code/datums/components/uplink.dm [hunk 2]
```diff
@@ -128,6 +129,15 @@
 	if(istype(item, /obj/item/stack/telecrystal))
 		load_tc(user, item)
 
+/datum/component/uplink/proc/on_examine(datum/souuce, mob/user, list/examine_list)
```
Comment (B4CKU) [inline code/datums/components/uplink.dm:132]: there's a small typo in the name of the "datum/source" variable

Comment (etherware-novice) [reply code/datums/components/uplink.dm:132]: well its not like the vars used anyway

Comment (dragomagol) [reply code/datums/components/uplink.dm:132]: C'mon

File: code/datums/components/uplink.dm [hunk 3]
```diff
@@ -128,6 +129,15 @@
 	if(istype(item, /obj/item/stack/telecrystal))
 		load_tc(user, item)
 
+/datum/component/uplink/proc/on_examine(datum/source, mob/user, list/examine_list)
+	if(user != owner)
```
Comment (ZephyrTFA) [inline code/datums/components/uplink.dm:133]:
```suggestion
	SIGNAL_HANDLER

	if(user != owner)
```

File: code/datums/components/uplink.dm [hunk 4]
```diff
@@ -128,6 +129,17 @@
 	if(istype(item, /obj/item/stack/telecrystal))
 		load_tc(user, item)
 
+/datum/component/uplink/proc/on_examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	if(user != owner)
+		return
+	examine_list += span_warning("\The [parent] contains your hidden uplink.")
```
Comment (Mothblocks) [inline code/datums/components/uplink.dm:137]:
```suggestion
	examine_list += span_warning("[parent] contains your hidden uplink.")
```

"The" is implicit.

File: code/datums/components/uplink.dm [hunk 5]
```diff
@@ -128,6 +129,17 @@
...
+	if(unlock_code)
+		examine_list += span_warning("The code to unlock it is [unlock_code].")
```
Comment (Mothblocks) [inline code/datums/components/uplink.dm:139]: This should be in the same line as the "this contains your uplink" IMO. Then bold the unlock code with `span_bold`

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Nits

File: code/datums/components/uplink.dm [hunk 6]
```diff
@@ -128,6 +129,17 @@
 	if(istype(item, /obj/item/stack/telecrystal))
 		load_tc(user, item)
 
+/datum/component/uplink/proc/on_examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	if(user != owner)
+		return
+	examine_list += span_warning("[parent] contains your hidden uplink\
+	[unlock_code ? ", and the code to unlock it is [span_boldwarning(unlock_code)]" : null].")
```
Comment (Mothblocks) [inline code/datums/components/uplink.dm:138]:
```suggestion
		[unlock_code ? ", the code to unlock it is [span_boldwarning(unlock_code)]" : null].")
```

**Pull request merged**